salt/common/tools/sbin/so-status

```diff
@@ -170,78 +176,70 @@ parse_status() {
-        state_color="\e[1;32m"
-    elif [[ $service_state = "$PENDING_STRING" ]] || [[ $service_state = "$DISABLED_STRING" ]] || [[ $service_state = "$STARTING_STRING" ]] || [[ $service_state = "$WAIT_START_STRING" ]]; then
-        state_color="\e[1;33m"
-        EXITCODE=2
+    is_tty
+    local service_name="${1}"
+    local service_state="" ; service_state="$( parse_status "${1}" "${2}" )"
+    # XXX: What will we do if tput isn't avalable?
+    local line=""
+    if (( __tty == 1 )); then
+        local NC; NC="$(tput sgr0)" # no color
+        local red=""; red="$(tput setaf 1 bold)"
+        local green; green="$(tput setaf 2 bold)"
+        local yellow; yellow="$(tput setaf 3 bold)"
+
+        # construct a line of '------' so that the names and states are all aligned
+        local PADDING_CONSTANT=16 # two tabs wide
+        PADDING_CONSTANT=$((PADDING_CONSTANT + PADDING_CONSTANT / 2))
+        local columns ; columns=$(tput cols)
+        linewidth=$(( columns - PADDING_CONSTANT - ${#service_name} - ${#service_state} ))
+
+        for i in $(seq 0 "${linewidth}"); do
+          line="${line}-"
+        done
     fi
 
-    printf "    $service_name "
-    for i in $(seq 0 $(( $columns - $PADDING_CONSTANT - ${#service_name} - ${#service_state} ))); do
-        printf "${state_color}%b\e[0m" "-"
-    done
-    printf " [ "
-    printf "${state_color}%b\e[0m" "$service_state"
-    printf "%s    \n" " ]"
-}
-
-non_term_print_line() {
-    local service_name=${1}
-    local service_state="$( parse_status ${1} ${2} )"
-
-    if [[ $service_state = "$ERROR_STRING" ]] || [[ $service_state = "$MISSING_STRING" ]]; then
-        if [[ "$EXITCODE" -eq 0 ]]; then
-            EXITCODE=1
-        fi
-    elif [[ $service_state = "$PENDING_STRING" ]] || [[ $service_state = "$DISABLED_STRING" ]] || [[ $service_state = "$STARTING_STRING" ]] || [[ $service_state = "$WAIT_START_STRING" ]]; then
-        EXITCODE=2
+    if [[ $service_state = "$ERROR_STRING" ]] \
+        || [[ $service_state = "$MISSING_STRING" ]]; then
+          state_color="${red:-}"
+          if [[ "$EXITCODE" -eq 0 ]]; then
+              EXITCODE=1
+          fi
+    elif [[ $service_state = "$SUCCESS_STRING" ]]; then
+        state_color="${green:-}"
+    elif [[ $service_state = "$PENDING_STRING" ]] \
+        || [[ $service_state = "$DISABLED_STRING" ]] \
+        || [[ $service_state = "$STARTING_STRING" ]] \
+        || [[ $service_state = "$WAIT_START_STRING" ]]; then
+            state_color="${yellow:-}"
+            EXITCODE=2
     fi
 
-    printf "    $service_name "
-    for i in $(seq 0 $(( 35 - ${#service_name} - ${#service_state} ))); do
-        printf "-"
-    done
-    printf " [ "
-    printf "$service_state"
-    printf "%s    \n" " ]"
+    service_state="${state_color:-}${service_state}${NC:-}"
+    line="${state_color:-}${line:-}${NC:-}"
+    printf "\t%s %s [ %s ]\n" "${service_name}" "${line:-}" "${service_state}"
```

Comment:
Much easier to read this code, this makes the function much simpler to maintain.